src/transport/manager/mod.rs

```diff
@@ -267,13 +267,13 @@ impl TransportManager {
         let peers = Arc::new(RwLock::new(HashMap::new()));
         let (cmd_tx, cmd_rx) = channel(256);
         let (event_tx, event_rx) = channel(256);
-        let listen_addresses = Arc::new(RwLock::new(HashSet::new()));
+        let listen_addresses = PublicAddresses::new(local_peer_id);
```

Comment:
Again, it looks like there is some confusion between public addresses and listen addresses.